src/components/Portal/Portal.jsx

```diff
@@ -44,47 +26,55 @@ const Portal = createClass({
 		children: node`
 			any valid React children
 		`,
+
 		className: any`
 			Appended to the component-specific class names set on the root element.
 			Value is run through the \`classnames\` library.
 		`,
 
-		portalId: string.isRequired`
+		portalId: string`
 			The \`id\` of the portal element that is appended to \`document.body\`.
 		`,
 	},
-	contextTypes: {
-		store: any, // access `store` from the context object
+
+	getInitialState() {
+		return { isReady: false };
 	},
-	render: () => null,
+
 	componentDidMount() {
 		const { portalId } = this.props;
 
-		let portalElement = window.document.getElementById(portalId);
+		let portalElement;
+
+		if (portalId) {
+			portalElement = document.getElementById(portalId);
+		}
 		if (!portalElement) {
-			portalElement = window.document.createElement('div');
+			this.manuallyCreatedPortal = true;
+			portalElement = document.createElement('div');
 			portalElement.id = portalId;
-			window.document.body.appendChild(portalElement);
+			document.body.appendChild(portalElement);
 		}
 		this.portalElement = portalElement;
-		this.componentDidUpdate();
+		this.setState({ isReady: true });
 	},
 	componentWillUnmount() {
-		ReactDOM.unmountComponentAtNode(this.portalElement);
-		window.document.body.removeChild(this.portalElement);
+		if (this.manuallyCreatedPOrtal) {
```

Comment:
`manuallyCreatedPOrtal` looks misspelled

Reply:
whoops, will fix this

Reply:
fixed and added a test case